src/ipc/ipc.c

```diff
@@ -436,49 +436,34 @@ int ipc_comp_dai_config(struct ipc *ipc, struct sof_ipc_dai_config *config)
 	struct sof_ipc_comp_dai *dai;
 	struct ipc_comp_dev *icd;
 	struct list_item *clist;
-	struct comp_dev *dev;
 	int ret = -ENODEV;
 
 	/* for each component */
 	list_for_item(clist, &ipc->shared_ctx->comp_list) {
 		icd = container_of(clist, struct ipc_comp_dev, list);
-		switch (icd->type) {
-		case COMP_TYPE_COMPONENT:
+		/* make sure we only config DAI comps */
+		if (icd->type != COMP_TYPE_COMPONENT)
+			continue;
 
-			/* make sure we only config DAI comps */
-			switch (icd->cd->comp.type) {
-			case SOF_COMP_DAI:
-			case SOF_COMP_SG_DAI:
-				dev = icd->cd;
-				dai = (struct sof_ipc_comp_dai *)&dev->comp;
-
-				/*
-				 * set config if comp dai_index matches
-				 * config dai_index.
-				 */
-				if (dai->dai_index == config->dai_index &&
-				    dai->type == config->type) {
-					ret = comp_dai_config(dev, config);
-					if (ret < 0) {
-						trace_ipc_error
-						("ipc_comp_dai_config() "
-						"error: comp_dai_config() "
-						"failed");
-						return ret;
-					}
-				}
-				break;
-			default:
-				break;
+		if (icd->cd->comp.type == SOF_COMP_DAI ||
+		    icd->cd->comp.type == SOF_COMP_SG_DAI) {
+			dai = (struct sof_ipc_comp_dai *)&icd->cd->comp;
+			/*
+			 * set config if comp dai_index matches
+			 * config dai_index.
+			 */
+			if (dai->dai_index == config->dai_index &&
+			    dai->type == config->type) {
+				ret = comp_dai_config(icd->cd, config);
 			}
-
-			break;
-		/* ignore non components */
-		default:
-			break;
 		}
 	}
 
+	if (ret < 0) {
+		trace_ipc_error("ipc_comp_dai_config() error: "
+				"comp_dai_config() failed");
```

Comment:
No need for curly braces around trace_ipc_error here.

Reply:
I need because it is a macro

Reply:
The trace_* macros have been written to act like functions from the point of view of using them so no, the braces aren't needed.

Reply:
@ktrzcinx do you mean that if we have something like this
```
if (ret < 0) 
    trace_();
foo();
```

and the ``trace_()`` expands to nothing then we are in big trouble?

Reply:
I'm OK with the code as it is now, but we should keep this discussion open.

Reply:
I claim that when `trace_*` expands to nothing then it will be ok, but when `trace_*` expand to any code with more than one line of code without `{ } `  inside then it won't. It will be need to spend much more time to debug what happen in such a situation in comparison with usage redundant parenthesis. I know that such a macro function is not a good solution but it is open source project and when using extra `{}` I know that I am safe for such a situation. Such a situation is not a big abstraction, also in tracing functions, for example: 
```
#define trace(obj_ptr,msg,...) \ 
   typeof(obj_ptr) __obj_ptr__ = obj_ptr; \
   _log_fun(__obj_ptr__ ->id, __obj_ptr__ ->parent_id, __VA_ARGS__)
```

Now 
```
if (ret < 0)
   trace(obj,"blah");
foo();
```

will trace independently ret value but my solution will still work correctly. In conclusion, I rather prefer defensive programming with extra `{}` from debugging.